src/main/kotlin/com/linchpino/ai/model/Prompt.kt

```diff
@@ -27,26 +34,13 @@ public class Prompt {
             ]
         }
         Provide me response in json without any other information.
-        """;
-
-    private final RequestDetail requestDetail;
 
-    public Prompt(RequestDetail requestDetail) {
-        this.requestDetail = requestDetail;
-    }
+        """.trimIndent()
 
-    public static Prompt of(RequestDetail requestDetail) {
-        if (requestDetail == null) {
-            throw new IllegalArgumentException("Request detail cannot be null");
+        @JvmStatic
+        fun of(requestDetail: RequestDetail?): Prompt {
```

Comment:
this method does nothing other than checking for null, and you should avoid passing null in the first place. you can easily remove this method